app/src/main/java/zapsolutions/zap/fragments/SettingsFragment.java

```diff
@@ -305,10 +305,16 @@ public void onClick(DialogInterface dialog, int whichButton) {
 
 
     private CharSequence[] joinCharSequenceArrays(CharSequence[] first, CharSequence[] second) {
-        List<CharSequence> both = new ArrayList<CharSequence>(first.length + second.length);
-        Collections.addAll(both, first);
-        Collections.addAll(both, second);
-        return both.toArray(new CharSequence[both.size()]);
+        try {
+            List<CharSequence> both = new ArrayList<CharSequence>(first.length + second.length);
```

Comment:
To avoid the nullpointer, could we check `first` and `second` for `null` and if both are, return an empty array, otherwise only the filled one? 
I couldn't reproduce the bug so I'm not sure what exactly the problem is. Maybe you can give me a hint how to reproduce 👍

Reply:
First (btc units) is never null, second ( fiat units) is null if no exchange rate data has been fetched yet.
In that case only the first will be returned.

As the first never gets null we don't have to check that.

To reproduce the error you have to whipe your application data. Then prevent internet connection by going to airplain mode and turning off wifi. Then open the app and go to settings.

Reply:
I have now changed the function like you suggested as this makes it more general and not only work in exactly this situation :+1: